### PR TITLE
[10.0][IMP] base_view_inheritance_extension : Add new feature 'target_position'

### PR DESCRIPTION
**Pull request by remihb**
By default, inheritance with spec **position="move"** append the node inside the target.
The spec **target_position** let you choose what you want to do :

- **inside**: append the node inside the target
- **before**: append the node before target
- **after**: append the node after target

By default, if no value is provide we use the standard behavior **inside**.

### CODE REVIEW

**Comment by remihb** [comment]
As #1056 in 9.0 but for version 10.0 with adding tests.

File: base_view_inheritance_extension/README.rst [hunk 1]
```diff
@@ -33,10 +33,18 @@ Move an element in the view
...
 .. code-block:: xml
 
-    <xpath expr="$xpath" position="move" target="$targetxpath" />
+    <xpath expr="$xpath" position="move" target="$targetxpath" target_position="$target_position"/>
 
-This can also be used to wrap some element into another, create the target
-element first, then move the node youwant to wrap there.
+Availables values for target_position attribute:
+
+* inside (default if omited)
```
Comment (hbrunn) [inline base_view_inheritance_extension/README.rst]: omitTed

File: base_view_inheritance_extension/models/ir_ui_view.py [hunk 2]
```diff
@@ -125,7 +125,21 @@ def inheritance_handler_xpath(self, source, specs, inherit_id):
         target_node = self.locate_node(
             source, etree.Element(specs.tag, expr=specs.get('target'))
         )
-        target_node.append(node)
+
+        target_position = specs.get('target_position') or 'inside'
```
Comment (hbrunn) [inline base_view_inheritance_extension/models/ir_ui_view.py]: pass `'inside'` as second parameter to `get`

**Review by hbrunn — CHANGES_REQUESTED** [review]
thanks, only small things found

File: base_view_inheritance_extension/README.rst (hunk 1 above)
Comment (remihb) [reply base_view_inheritance_extension/README.rst]: > omitted

 OK

File: base_view_inheritance_extension/models/ir_ui_view.py (hunk 2 above)
Comment (remihb) [reply base_view_inheritance_extension/models/ir_ui_view.py]: > target_position = specs.get('target_position', 'inside')

OK

**Review by hbrunn — APPROVED** [review]

File: base_view_inheritance_extension/models/ir_ui_view.py [hunk 3]
```diff
@@ -125,7 +125,21 @@ def inheritance_handler_xpath(self, source, specs, inherit_id):
...
+            target_node.addnext(node)
+        elif target_position == 'before':
+            target_node.addprevious(node)
+        elif target_position == 'inside':
+            target_node.append(node)
+        else:
+            self.raise_view_error(
+                _("Invalid target_position attribute: '%s'") % target_position,
+                inherit_id,
+                context=self.env.context
```
Comment (hbrunn) [inline base_view_inheritance_extension/models/ir_ui_view.py]: this needs to go away: https://github.com/OCA/OCB/blob/10.0/odoo/addons/base/ir/ir_ui_view.py#L460

**Review by hbrunn — CHANGES_REQUESTED** [review]

File: base_view_inheritance_extension/models/ir_ui_view.py (hunk 3 above)
Comment (remihb) [reply base_view_inheritance_extension/models/ir_ui_view.py]: I don't understand, I have to remove the **raise** itself or to raise in another way ?

Comment (hbrunn) [reply base_view_inheritance_extension/models/ir_ui_view.py]: I mean the context parameter, that's not in the function signature

Comment (remihb) [reply base_view_inheritance_extension/models/ir_ui_view.py]: Ah sure, it was late, I misread the code ^^'

**Review by hbrunn — APPROVED** [review]

**Comment by remihb** [comment]
Any news for this PR ?

**Review by thomaspaulb — APPROVED** [review]
Really well done, including the demo data and the tests. Tested this in an installation and it works.

You could still mention your name in the 'contributors' section of the README.

**Comment by hbrunn** [comment]
/ocabot merge

thanks! Note however that we've deprecated this in favor of the Odoo way of doing it for later versions: https://github.com/OCA/server-tools/blob/12.0/base_view_inheritance_extension/README.rst#usage

**Comment by remihb** [comment]
> This PR looks fantastic, let's merge it!

Cool :+1: 

> thanks! Note however that we've deprecated this in favor of the Odoo way of doing it for later versions: https://github.com/OCA/server-tools/blob/12.0/base_view_inheritance_extension/README.rst#usage

Sure, but still needed for older versions :wink:

**Pull request merged**